src/seq/bioseq.jl

```diff
@@ -1027,3 +1027,73 @@ end
 function enc64{A}(::BioSequence{A}, x)
     return UInt64(encode(A, convert(eltype(A), x)))
 end
+
+
+# Sequences to Matrix
+# -------------------
+
+"""
+    seqmatrix{A<:Alphabet}(vseq::Vector{BioSequence{A}}, major=:seq|:site)
+
+Construct a matrix of nucleotides or amino acids from a vector of `BioSequence`s.
+
+If parameter major is set to `:site`, the matrix is created such that one
+nucleotide from each sequence is placed in each column i.e. the matrix is laid
+out in site-major order.
+This means that iteration over one position of many sequences is efficient,
+as julia arrays are laid out in column major order.
+
+If the parameter major is set to `:seq`, the matrix is created such that each
+sequence is placed in one column i.e. the matrix is laid out in sequence-major
+order.
+This means that iteration over each sequence is efficient,
+as julia arrays are laid out in column major order.
+
+# Examples
+```julia
+julia> seqs = [dna"AAA", dna"TTT", dna"CCC", dna"GGG"]
+4-element Array{Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},1}:
+ 3nt DNA Sequence:
+AAA
+ 3nt DNA Sequence:
+TTT
+ 3nt DNA Sequence:
+CCC
+ 3nt DNA Sequence:
+GGG
+
+julia> seqmatrix(seqs, :site)
+4x3 Array{Bio.Seq.DNANucleotide,2}:
+ DNA_A  DNA_A  DNA_A
+ DNA_T  DNA_T  DNA_T
+ DNA_C  DNA_C  DNA_C
+ DNA_G  DNA_G  DNA_G
+
+ julia> seqmatrix(seqs, :seq)
+ 3x4 Array{Bio.Seq.DNANucleotide,2}:
+  DNA_A  DNA_T  DNA_C  DNA_G
+  DNA_A  DNA_T  DNA_C  DNA_G
+  DNA_A  DNA_T  DNA_C  DNA_G
+```
+"""
+function seqmatrix{A<:Alphabet}(vseq::Vector{BioSequence{A}}, major=:site)
+    if major == :site
+        nsites = minimum([length(seq) for seq in vseq])
```

Comment:
I'll make it more strict and require sequences of equal length - I think the more predictable behaviour the better, otherwise if someone doesen't remember to check in their calling method or script, depending on what they're doing they might get burned.